### PR TITLE
Migrate to modules

### PR DESCRIPTION
**Pull request by tonglil**
Migrate to use go modules.

### CODE REVIEW

File: .drone.yml [hunk 1]
```diff
@@ -21,8 +22,10 @@ pipeline:
   test:
     image: golang:1
     environment:
+      - GO111MODULE=on
```
Comment (fsouza) [inline .drone.yml]: Can probably add ``GOPROXY=https://proxy.golang.org`` to speed things up :D

File: .drone.yml [hunk 2]
```diff
@@ -21,8 +22,10 @@ pipeline:
...
       - CGO_ENABLED=0
     commands:
+      - go mod download
       - go vet
       - go test -cover
```
Comment (fsouza) [inline .drone.yml]: Not part of this PR, but in recent versions of Go `go test` and `go vet` can be combined with `go test -vet all`.

**Comment by tonglil** [comment]
> should we migrate to cloud.drone.io?

Yes, that is next (figure I do them separately to not break anything).
